### DISCUSSION TITLE
en/getting-started/quick-start/

### DISCUSSION THREAD

**Discussion by giscus** [go-gost/gost#417] [General]
# en/getting-started/quick-start/

https://gost.run/en/getting-started/quick-start/

**Comment by Maxx-Yang** [comment]
With this command `gost -L=tcp://:8080/192.168.1.1:80 -F socks5://192.168.1.2:1080`, I want to know the setup on the '192.168.1.1' server, Is the command `gost -L=:8080` ok?

**Reply by gabri-ns** [reply]
no... you should run it from 192.168.1.2 as that is the one mentioned by **-F**.

your command is effectively running a port forwarding from your local tcp port 8080 to 192.168.1.1:80 via 192.168.1.2. so when you open browser to http://127.0.0.1:8080/ it will be actually accessing http://192.168.1.1/

**Comment by 510846** [comment]
Gost V2.x 是支持在 json配置文件中用“简写”方式，如：
```
{
  "services": [
    "tcp://:8123/192.168.100.1:80"
  ]
}
```

在 V3.x 中是否可以支持这种“简写”方式呢？ 
*** 我的大部分都是这种简单转发，按照常规的写法，10条就要书写 100多行配置了 ***

**Reply by cnwhy** [reply]
可以用API来管理配置，就不用在乎配置文件是多少行了
试试这个 [gost-ui](https://github.com/cnwhy/gost-ui)

**Comment by beck-8** [comment]
更详细的使用说明请参考端口转发。
https://gost.run/getting-started/tutorials/port-forwarding/
url 404

**Comment by laiqun** [comment]
本地端口转发和远程端口转发是两种不同的端口转发方式，主要用于通过SSH隧道实现网络流量的转发。以下是它们的区别：

### 本地端口转发（Local Port Forwarding）
- **定义**：将本地机器的某个端口转发到远程机器的指定端口。
- **用途**：访问远程服务或绕过防火墙限制。
- **命令格式**：
  ```bash
  ssh -L [本地IP:]本地端口:目标IP:目标端口 用户名@远程主机
  ```
- **示例**：
  ```bash
  ssh -L 8080:192.168.1.100:80 user@remotehost
  ```
  - **解释**：将本地的8080端口转发到远程主机`remotehost`的192.168.1.100的80端口。
  - **访问方式**：在本地浏览器访问`http://localhost:8080`，实际访问的是`192.168.1.100:80`。

### 远程端口转发（Remote Port Forwarding）
- **定义**：将远程机器的某个端口转发到本地机器的指定端口。
- **用途**：允许外部网络访问本地服务。
- **命令格式**：
  ```bash
  ssh -R [远程IP:]远程端口:目标IP:目标端口 用户名@远程主机
  ```
- **示例**：
  ```bash
  ssh -R 8080:localhost:80 user@remotehost
  ```
  - **解释**：将远程主机的8080端口转发到本地的80端口。
  - **访问方式**：在远程主机上访问`http://localhost:8080`，实际访问的是本地的80端口。

### 主要区别
1. **方向**：
   - **本地端口转发**：从本地到远程。
   - **远程端口转发**：从远程到本地。

2. **用途**：
   - **本地端口转发**：访问远程服务。
   - **远程端口转发**：暴露本地服务给远程网络。

3. **命令参数**：
   - **本地端口转发**：`-L`。
   - **远程端口转发**：`-R`。

### 应用场景
- **本地端口转发**：访问远程数据库、Web服务等。
- **远程端口转发**：将本地开发环境暴露给外部网络，方便远程调试。

总结来说，本地端口转发用于访问远程服务，远程端口转发用于暴露本地服务给远程网络。

**Reply by beck-8** [reply]
说的很好，下次别说了

**Comment by pianlecsapp** [comment]
请问可以讲本地udp端口数据封装成wensockets数据转发吗

**Comment by saysth7** [comment]
远程转发tcp端口的示例里，说明中咋没提到1080呢？

**Reply by saysth7** [reply]
对了，问的三家AI都说是在本机上的2222端口：kimi,deepseek,doubao

**Reply by saysth7** [reply]
千问的解释看起来才是对的。都是网页版默认选项。
